molgenis-data-annotators/src/main/java/org/molgenis/data/annotation/filter/MultiAllelicResultFilter.java

```diff
@@ -118,12 +133,110 @@ private void filter(Entity annotatedEntity, List<Entity> processedResults, Entit
                     newAttributeValue.append(".");
                 }
             }
-            // add entity only if something was found
-            if (!newAttributeValue.toString().equals("."))
+            // add entity only if something was found, so no '.' or any multiple of '.,' (e.g. ".,.,.")
+            if (!newAttributeValue.toString().matches("[\\.,]+"))
             {
                 entity.set(attributeMetaData.getName(), newAttributeValue.toString());
                 processedResults.add(entity);
             }
         }
 	}
+	
+	/**
+	 * Combine ALT information per reference allele (in VCF there is only 1
+	 * reference by letting ALT vary, but that might not always be the case)
+	 * 
+	 * So we want to support this hypothetical example:
+	 * 3	300	G	A	0.2|23.1
+	 * 3	300	G	T	-2.4|0.123
+	 * 3	300	G	X	-0.002|2.3
+	 * 3	300	G	C	0.5|14.5
+	 * 3	300	GC	A	0.2|23.1
+	 * 3	300	GC	T	-2.4|0.123
+	 * 3	300	C	GX	-0.002|2.3
+	 * 3	300	C	GC	0.5|14.5
+	 * 
+	 * and it should become:
+	 * 
+	 * 3	300	G	A,T,X,C	0.2|23.1,-2.4|0.123,-0.002|2.3,0.5|14.5
+	 * 3	300	GC	A,T	0.2|23.1,-2.4|0.123
+	 * 3	300	C	GX,GC	-0.002|2.3,0.5|14.5
+	 * 
+	 * so that the multi-allelic filter can then find back the appropriate values as if it were a multi-allelic VCF line
+	 * 
+	 */
+	public Iterable<Entity> merge(Iterable<Entity> resourceEntities)
```

Comment:
A unit test for this method would be cool. 
Testing this through the entire annotator path is a bit tricky IMO